### DISCUSSION TITLE
SuperResolution samples

### DISCUSSION THREAD

**Discussion by sn4k3** [emgucv/emgutf#90] [Q&A]
Is there any sample using SuperResolution models such as [esrgan](https://www.kaggle.com/models/kaggle/esrgan-tf2/frameworks/tensorFlow2)?

**Comment by emgucv** [comment]
If you use Emgu CV here is a supper resolution model that you can use:
https://github.com/emgucv/emgucv/blob/master/Emgu.CV.Models/Dnn/Superres.cs

It is part of the 4.8.1 release.

**Reply by sn4k3** [reply]
I tested that before, but not only the result is not optimal, it takes a ton of ram and time to compute a "simple" 1290x871 image (2m22s):

```C#
        using var mat = CvInvoke.Imread("D:\\shark.jpg");
        using var ss = new DnnSuperResImpl("edsr", 4);

        ss.ReadModel("EDSR_x4.pb");
        ss.Upsample(mat, mat);

        CvInvoke.Imwrite("D:\\shark_up.jpg", mat);
```

It upscales, but the quality remains the same.

RAM:
![image](https://github.com/emgucv/emgutf/assets/113281/8dc1630e-89de-4415-aea2-37ce29a645eb)
7GB RAM for such operation doesn't look right to me (I also can be wrong in assumption that this operation should run in less ram)

Original VS upscaled (No quality improves, only resolution had grown)
![image](https://github.com/emgucv/emgutf/assets/113281/54ebf583-01cb-4139-94d6-33344183c1d2)


As so I need a more optimized algorithms and also would like to play with some tensors like super sample, denoiser, deblur, sharper and others to improve images. However, all samples look like to infer with data and obtain the predictions. Would be nice to have a sample on how to apply simple image models. In python they look and run like magic but in C# TF I have no clue how to call and apply.
